core/server/server.js

```diff
@@ -28,7 +28,7 @@ const requiredUrl = optionalUrl.required()
 
 const publicConfigSchema = Joi.object({
   bind: {
-    port: Joi.number().port(),
+    port: Joi.optional(),
```

Comment:
Does the below work in the App Service Windows plan env? I'd prefer it if so to keep more explicit validation while supporting that use case

```suggestion
    port: Joi.alternatives().try(
      Joi.number().port(),
      Joi.string(),
    ),
```

Reply:
We could also try using https://www.npmjs.com/package/unc-path-regex.

Reply:
That should work fine. I've got a unit test that confirms it as well.

Reply:
@paulmelnikow were you thinking of using `unc-path-regex` as part of a custom `@hapi/joi` validator vs the more lenient `Joi.string()`?

Reply:
(unresolving this conversation as there were still open questions)

Named pipes happen to be a very specific subset of Windows file paths, the aforementioned `unc-path-regex` will accept them but is much more general. In my opinion, adding a dependency for a single edge-case check on startup would be overkill, especially if that dependency does not do exactly what we want.

Why not write a simple regex that validates against the Windows pipe naming specs? The [Microsoft docs](https://docs.microsoft.com/en-us/windows/win32/ipc/pipe-names) indicates `\\ServerName\pipe\PipeName` and even `\\.\pipe\PipeName` if it's the same server.

Reply:
If we use a regex we can use `Joi.string().regex()`; we don't need to make an extension. That package is basically one line, though we could add our own regex that is more limited to `\\.\pipe\PipeName`.